spec/puppet/resource_api_spec.rb

```diff
@@ -1344,6 +1430,7 @@ def set(_context, changes)
         context 'when flushing' do
           before(:each) do
             Puppet.debug = true
+            instance.my_provider.set(nil, nil) # reset the current_state
```

Comment:
Huh, how are the changes above leaking into this test case?

Reply:
believe random seed order was changed when the new test went in and exposed the bug

Reply:
ah, ok! then please put this in a separate commit with a short note about that, to inform future-us of what went down here.